sklearn/inspection/_partial_dependence.py

```diff
@@ -5,159 +5,33 @@
-    points between the percentiles of the jth column of X.
-    If ``grid_resolution`` is bigger than the number of unique values in the
-    jth column of X, then those unique values will be used instead.
-
-    Parameters
-    ----------
-    X : ndarray, shape (n_samples, n_target_features)
-        The data
-
-    percentiles : tuple of floats
-        The percentiles which are used to construct the extreme values of
-        the grid. Must be in [0, 1].
-
-    grid_resolution : int
-        The number of equally spaced points to be placed on the grid for each
-        feature.
-
-    Returns
-    -------
-    grid : ndarray, shape (n_points, n_target_features)
-        A value for each feature at each point in the grid. ``n_points`` is
-        always ``<= grid_resolution ** X.shape[1]``.
-
-    values : list of 1d ndarrays
-        The values with which the grid has been created. The size of each
-        array ``values[j]`` is either ``grid_resolution``, or the number of
-        unique values in ``X[:, j]``, whichever is smaller.
-    """
-    if not isinstance(percentiles, Iterable) or len(percentiles) != 2:
-        raise ValueError("'percentiles' must be a sequence of 2 elements.")
-    if not all(0 <= x <= 1 for x in percentiles):
-        raise ValueError("'percentiles' values must be in [0, 1].")
-    if percentiles[0] >= percentiles[1]:
-        raise ValueError('percentiles[0] must be strictly less '
-                         'than percentiles[1].')
-
-    if grid_resolution <= 1:
-        raise ValueError("'grid_resolution' must be strictly greater than 1.")
-
-    values = []
-    for feature in range(X.shape[1]):
-        uniques = np.unique(_safe_indexing(X, feature, axis=1))
-        if uniques.shape[0] < grid_resolution:
-            # feature has low resolution use unique vals
-            axis = uniques
-        else:
-            # create axis based on percentiles and grid resolution
-            emp_percentiles = mquantiles(
-                _safe_indexing(X, feature, axis=1), prob=percentiles, axis=0
-            )
-            if np.allclose(emp_percentiles[0], emp_percentiles[1]):
-                raise ValueError(
-                    'percentiles are too close to each other, '
-                    'unable to build the grid. Please choose percentiles '
-                    'that are further apart.')
-            axis = np.linspace(emp_percentiles[0],
-                               emp_percentiles[1],
-                               num=grid_resolution, endpoint=True)
-        values.append(axis)
-
-    return cartesian(values), values
-
-
 def _partial_dependence_recursion(est, grid, features):
     return est._compute_partial_dependence_recursion(grid, features)
 
 
 def _partial_dependence_brute(est, grid, features, X, response_method):
     averaged_predictions = []
-
-    # define the prediction_method (predict, predict_proba, decision_function).
```

Comment:
To the reviewer: this is now _base._get_predictions

Reply:
This is now _partial_dependence._get_predictions as ICE was moved to _partial_dependence.py file.